### DISCUSSION TITLE
Integration of ABC

### DISCUSSION THREAD

**Discussion by nindanaoto** [bitwuzla/bitwuzla#128] [General] [answered]
Dear maintainers, 

 Hello, this is a question or feature request, not a bug report or something. 
 
I know that [STP](https://github.com/stp/stp) integrates [Berkley ABC](https://github.com/berkeley-abc/abc.git) to minimize the AIG representation of the problem.   
 Is there any plan to integrate ABC into Bitwuzla as well? If not, what made the strategy difference between Bitwuzla and STP, the two most advanced QF_BV SMT solvers, regarding integrating the AIG synthesis tool?

**Comment by mpreiner** [comment, marked as answer]
We don't have any plans to integrate ABC. Bitwuzla implements its own AIG representation and simplifications similar to what Boolector does. The simplifications are explained in [this paper](https://fmv.jku.at/papers/BrummayerBiere-MEMICS06.pdf).

In fact, a prior version of CVC4 had optional support for ABC, but we removed it since there were no significant performance gains.

**Reply by nindanaoto** [reply]
Thank you for moving my question to the correct place and giving me the paper information. 
Since I know that logic synthesis people are hardly working on AIG minimization like [this year's IWLS competition](https://github.com/alanminko/iwls2024-ls-contest), I thought that if we have the application of such tools to SMT solving, it seems to be an interesting topic. I can understand the such a situation might happen since AIG minimization is just a preprocessing in this case, so spending too much time to get minimal AIG with synth tools seems to be not the way to go...... Anyway, thank you for sharing your thoughts! If you have any further opinions, of course, I'm happy to hear that, but my question seems to be resolved.
